jvm/src/test/scala/tastyquery/PositionSuite.scala

```diff
@@ -0,0 +1,290 @@
+          |  class Inner
+          |}""".stripMargin,
+        "class Inner"
+      )
+    )
+  }
+
+  test("super") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Super")
+    assertEquals(collectCode[Super](tree, code), List("super", "super[Base]"))
+  }
+
+  test("self".ignore) {
+    // ignore because the span of Self is impossible to construct
+    val (tree, code) = unpickleWithCode(simple_trees / tname"ClassWithSelf")
+    assertEquals(collectCode[ValDef](tree, code), List("self: ClassWithSelf =>"))
+  }
+
+  /** Import and export */
+
+  test("import") {
+    val (tree, code) = unpickleWithCode(imports / tname"Import")
+    assertEquals(collectCode[Import](tree, code), List("import imported_files.A"))
+  }
+
+  test("import-selector-multiple") {
+    val (tree, code) = unpickleWithCode(imports / tname"MultipleImports")
+    assertEquals(collectCode[ImportSelector](tree, code), List("A", "B"))
+  }
+
+  test("import-selector-bound") {
+    val (tree, code) = unpickleWithCode(imports / tname"ImportGivenWithBound")
+    assertEquals(collectCode[ImportSelector](tree, code), List("A", "given A"))
+  }
+
+  test("import-selector-renamed") {
+    val (tree, code) = unpickleWithCode(imports / tname"RenamedImport")
+    assertEquals(collectCode[ImportSelector](tree, code), List("A => ClassA"))
+  }
+
+  test("export") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Export")
+    assertEquals(
+      collectCode[Export](tree, code),
+      List("export first.status", "export second.{status => _, *}", "export givens.given AnyRef")
+    )
+  }
+
+  /** Throw and try-catch */
+
+  test("throw") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"ThrowException")
+    assertEquals(collectCode[Throw](tree, code), List("throw new NullPointerException"))
+  }
+
+  test("try") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"TryCatch")
+    assertEquals(
+      collectCode[Try](tree, code),
+      List("""try {
+             |      ThrowException().f()
+             |      1
+             |    } catch {
+             |      case _ => 0
+             |    } finally {
+             |      ()
+             |    }""".stripMargin)
+    )
+  }
+
+  /** Types */
+
+  test("typed") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Typed")
+    assertEquals(collectCode[Typed](tree, code), List("1: Int"))
+  }
+
+  test("type-member") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"TypeMember")
+    assertEquals(
+      collectCode[TypeMember](tree, code),
+      List(
+        "type TypeMember = Int",
+        "type AbstractType",
+        "type AbstractWithBounds >: Null <: AnyRef",
+        "opaque type Opaque = Int",
+        "opaque type OpaqueWithBounds >: Null <: AnyRef = Null"
+      )
+    )
+  }
+
+  test("type-apply") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"TypeApply")
+    assertEquals(collectCode[TypeApply](tree, code), List("Seq"))
```

Comment:
would be good to get a test with explicit type parameters also

Reply:
test cases are added